helusers/user_utils.py

```diff
@@ -111,7 +113,80 @@ def convert_to_uuid(convertable, namespace=None):
     return generated_uuid
 
 
+def migrate_ad_user(user_id: str, payload: dict, oidc: bool = False):
+    """Migrate old AD user from Tunnistamo to Keycloak.
+
+    Migration logic is only run on certain conditions:
+    - Correct authentication method is used (AMR-claim)
+    - Email domain is correct
+    - User with the new UUID doesn't exist yet
+    - Old user is found by first name, last name and email
+    - Old user has username generated by helusers.utils.uuid_to_username
+
+    Instead of allowing a new user to be created the migration is done by
+    replacing the old user UUID with the one from the incoming token
+    payload. Logic which is run later should take care of updating other
+    user related fields.
+
+    Primary key is separate from the user UUID, so the user UUID can be
+    changed. This migration should therefore retain all the data related
+    to the user.
+
+    Migration logic only supports authentication methods from this package
+    and Python Social Auth pipeline helusers.defaults.SOCIAL_AUTH_PIPELINE.
+    This doesn't support migrating users which are using e.g. a different
+    pipeline for Python Social Auth (e.g. the default pipeline).
+    """
+    amr_to_migrate = getattr(settings, "HELUSERS_AD_MIGRATE_AMR", ["helsinkiad"])
+    domains_to_migrate = getattr(
+        settings, "HELUSERS_AD_MIGRATE_EMAIL_DOMAINS", ["@hel.fi"]
+    )
+
+    if not getattr(settings, "HELUSERS_AD_MIGRATE_ENABLED", False):
+        return
```

Comment:
Of course this doesn't really matter in practice, but personally I find it clearest when the 'enabled' condition is checked right at the start, before anything else is done.

```suggestion
    if not getattr(settings, "HELUSERS_AD_MIGRATE_ENABLED", False):
        return
        
    amr_to_migrate = getattr(settings, "HELUSERS_AD_MIGRATE_AMR", ["helsinkiad"])
    domains_to_migrate = getattr(
        settings, "HELUSERS_AD_MIGRATE_EMAIL_DOMAINS", ["@hel.fi"]
    )
```